protocol.go

```diff
@@ -46,6 +46,11 @@ const (
 	// streaming endpoints. However, bidirectional streams are only supported
 	// when combined with HTTP/2.
 	ProtocolGRPCWeb
+	// ProtocolGRPCWebText is a variant of ProtocolGRPCWeb that uses base64
+	// text encoding for the request and response bodies.
+	//
+	// This protocol is not supported on the server side.
```

Comment:
Do we even need an exported constant for it?

It would likely be trivial to implement as a separate, simpler middleware handler. Then we have the option to potentially move it to connect-go in the future, to enable grpc-web-text there. The `Transcoder` could always delegate to that middleware first since it should basically be free/pass-through for non-grpc-web-text requests. Then we don't need a protocol constant for it all since the rest of the logic in here doesn't need to worry about it.

Reply:
It would also be relatively easy to test that separate middleware by copying it over into the connect conformance `referenceserver` and then see if the JS grpc-web-client starts passing the server stream tests that it currently opts out of. We'd have to enable grpc-web-text in that client, but that's as simple as changing the [mode](https://github.com/connectrpc/conformance/blob/main/buf.gen.yaml#L20) to "grpcwebtext" and re-generating.

(Admittedly, that's not a way to unit test this repo's implementation of it. But the tests you've added here could still be used of course.)